aspnetcore/signalr/authn-and-authz.md

```diff
@@ -0,0 +1,119 @@
+
+### Cookie authentication
+
+In a browser-based app, cookie authentication allows your existing user credentials to automatically flow to SignalR connections. When using the browser client, no additional configuration is needed. If the user is logged in to your app, the SignalR connection automatically inherits this authentication.
+
+Cookie authentication is not recommended unless the app only needs to authenticate users from the browser client. When using the [.NET Client](xref:signalr/dotnet-client), the `Cookies` property can be configured in the `.WithUrl` call in order to provide a cookie. However, using cookie authentication from the .NET Client requires the app to provide an API to exchange authentication data for a cookie.
+
+### Bearer token authentication
+
+Bearer token authentication is the recommended approach when using clients other than the browser client. In this approach, the client provides an access token that the server validates and uses to identify the user. The details of bearer token authentication are beyond the scope of this document, but there are some things to note when using this form of authentication in SignalR. On the server, bearer token authentication is configured using the [JWT Bearer middleware](/dotnet/api/microsoft.extensions.dependencyinjection.jwtbearerextensions.addjwtbearer).
+
+In the JavaScript client, the token can be provided using the [`accessTokenFactory`](xref:signalr/configuration#configure-bearer-authentication) option.
+
+[!code-typescript[Configure Access Token](authn-and-authz/sample/wwwroot/js/chat.ts?range=63-65)]
+
+In the .NET client, there is a similar [`AccessTokenProvider`](xref:signalr/configuration#configure-bearer-authentication) property that can be used to configure the token:
+
+```csharp
+var connection = new HubConnectionBuilder()
+    .WithUrl("https://example.com/myhub", options =>
+    { 
+        options.AccessTokenProvider = () => _myAccessToken;
+    })
+    .Build();
+```
+
+> [!NOTE]
+> The access token function you provide is called before **every** HTTP request made by SignalR. If you need to renew the token in order to keep the connection active (because it may expire during the connection), do so from within this function and return the updated token.
+
+In standard Web APIs, bearer tokens are sent in an HTTP header. However, due to security limitations imposed by browsers, the browser client is unable to set these headers when using the WebSockets or Server-Sent Events transports. When using those transports, the token is transmitted as a query string parameter. In order to support this on the server, additional configuration is required:
```

Comment:
Can we shorten some of these ROs?

 However the browser client is unable to set bearer token headers when using the WebSockets or Server-Sent Events transports. The browser can't set set bearer token headers with WebSockets or Server-Sent Events transports for security reasons.

I don't like my rewrite but it's easier to read and MT.